### DISCUSSION TITLE
Seeding from shapefile

### DISCUSSION THREAD

**Discussion by ocecept** [OpenDrift/opendrift#1180] [Q&A] [answered]
Dear all;

I am using a shapefile to seed 100,000 particles along our shelf region. 

When we release the particles at a specific time, it works as expected and particles are evenly released along the shapefile region. 

Now, we tried to release the 100,000 particles in the shape region but along 15 days. Instead of the particles to be evenly released along the shape region at each time step, we get all particles been released at a specific point (see animation attached). The release point changes with time and in the end all shape region will be "filled" with particles each one released in a different time step. 

Here is the seeding part of the code: 

```
# Start time
starttime = datetime(2022, 1, 2, 0, 0, 0)

# Time range
timed = [starttime - timedelta(days=15), starttime]

# Open model
o = PlastDrift(loglevel=0)

# Add current field
o.add_reader([cmems])

# model config
o.set_config('drift:stokes_drift', False)
o.set_config('drift:current_uncertainty', 0.1)
o.set_config('general:coastline_action', 'stranding')

# Add seeds at different origin_marker
for idx in range(3):
    o.seed_from_shapefile(SHAPEFILE_DIR+SHAPEFILE_NAME[idx],
                          number=100000,
                          featurenum=[1], time=timed, origin_marker=idx)
# Running
out='test.nc'
o.run(time_step=timedelta(minutes=-180), duration=timedelta(days=15), time_step_output=timedelta(minutes=720), outfile=out)
```
What we want is to have particles evenly (or randomly)  released in the space at each time step. Am I doing anything wrong? 

I am sending the animation and the figure with the final position at the end of the simulation. Our code is all attached. 

[setup.txt](https://github.com/OpenDrift/opendrift/files/13217725/setup.txt)
![oceanwnd3](https://github.com/OpenDrift/opendrift/assets/54149706/1456db17-d202-4acc-9770-5d1dce30a72a)
![oceanwnd3_animation](https://github.com/OpenDrift/opendrift/assets/54149706/c132330f-c938-40d4-90bf-4c30472defc6)

**Comment by knutfrode** [comment, marked as answer]
Hi,
Seeding within a polygon/shapefile is only supposed to work for specific times, so OpenDrift should have thrown an error here. I am in fact surprised that it did not crash.
Rather, you must do repeated seeding for each shapefile for each desired release time. Otherwise it would be very difficult for OpenDrift to know how you would like your particles to be distributed in time and space.

**Reply by ocecept** [reply]
Thanks a lot. I will do the repeated seeding as suggested.
